src/gamma.jl

```diff
@@ -764,3 +764,27 @@ const lfactorial = lfact
 export lfactorial
 
 end # @static if
+
+"""
+    lbinomial(n, k) = log(abs(binomial(n, k)))
+
+Accurate natural logarithm of the absolute value of the [`binomial`](@ref)
+coefficient `binomial(n, k)` for large `n` and `k` near `n/2`.
+"""
+function lbinomial(n::T, k::T) where {T<:Integer}
+    S = float(T)
+    (k < 0) && return typemin(S)
+    if n < 0
+       n = -n + k - 1
+    end
+    k > n && return typemin(S)
+    (k == 0 || k == n) && return zero(S)
+    (k == 1) && return log(abs(n))
+    if k > (n>>1)
+       k = n - k
+    end
+    -log1p(n) - lbeta(n - k + one(T), k + one(T))
+end
+lbinomial(n::Integer, k::Integer) = lbinomial(promote(n, k)...)
+
+export lbinomial
```

Comment:
This should go with the other exports.

Reply:
I put it now with `cosint`. Not sure if this is the "normal" place or not.